enhancements/sig-architecture/14-addon-cluster-proxy/README.md

```diff
@@ -0,0 +1,545 @@
+    replicas: 2
+    inClusterService:
+      namespace: open-cluster-management
+      name: proxy-server-default
+  proxyAgent:
+    image: <...>
+    replicas: 2
+status:
+  lastObserverdGeneration: 2
+  certificates:
+    signedCertificate: <...> # PEM-encoded
+    privateKey: <...>  # PKCS#8 RSA generated key
+    bootstrapSecret: <...>
+  conditions:
+  - type: "AllHealthy" 
+    status: "True"
+  - type: "AllUpToDate" 
+    status: "True"
+  components:
+    proxyServer:
+     - name: default-1
+       namespace: open-cluster-management
+       healthy: true
+       restartCount: 0
+     - name: default-2
+       namespace: open-cluster-management
+       healthy: true
+       restartCount: 0
+    proxyAgent:
+      - cluster: <deployed cluster name> 
+        lastUpdateTimestamp: <...>
+```
+
+
+##### ClusterManagementAddon
+
+`ClusterManagementAddon` is the original addon custom resource provided by the
+addon framework. Upon creating the addon resource and connecting it w/ an
+existing `ManagedProxyConfiguration` resource, the operator will be actively 
+syncing up the deployment of the proxy-servers to the configuration in the 
+following steps:
+
+1. Signing all the required X509 certificates and keys then stores them into
+   a secret object:
+
+   i.   a CSR requesting server certificate which will be shared across the
+   replicas of the proxy servers.
+
+   ii.  a CSR requesting client certificate for logging onto in the tunnel
+   (skipped if the proxy server under "uds" mode).
+   
+   iii. a bootstrap secret that allows the future proxy-agents to issue
+   CSR requests.
+
+2. Rolling/scaling out the proxy servers upon spec changes.
+
+3. Actively probing/reporting the status of the proxy server instances.
+
+And an example of `ClusterManagementAddon` is shown below:
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddon
+metadata:
+ name: apiserver-network-proxy
+spec:
+ addonMeta:
+   displayName: apiserver-network-proxy
+   description: "https://github.com/kubernetes-sigs/apiserver-network-proxy"
+ addonConfiguration:
+   # Importing templates for creating real agent instances.
+   crd: managedproxyconfigurations.proxy.addon.open-cluster-management.io
+   name: default
+```
+
+##### ManagedClusterAddon
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddon
+metadata:
+  name: apiserver-network-proxy
+  namespace: <deploying cluster name>
+spec:
+  relatedObjects:
+    - apiVersion: addon.open-cluster-management.io/v1alpha1
+      resource: clustermanagementaddons
+      name: apiserver-network-proxy
+  addonMeta:
+    displayName: apiserver-network-proxy
+    description: "https://github.com/kubernetes-sigs/apiserver-network-proxy"
+  addonConfiguration:
+    # Importing templates for creating real agent instances.
+    crd: managedproxyconfigurations.proxy.addon.open-cluster-management.io
+    name: default
+```
+
+##### CredentialProjection
```

Comment:
let's remove this  section

Reply:
done, also eliminated connecting section in the goal list